views-react/src/main/java/io/micronaut/views/react/ReactJSBeanFactory.java

```diff
@@ -62,4 +68,84 @@ Engine engine(ReactViewsRendererConfiguration configuration) {
             .sandbox(sandbox ? SandboxPolicy.CONSTRAINED : SandboxPolicy.TRUSTED)
             .build();
     }
+
+    @ReactBean
+    @Singleton
+    BeanPool<ReactJSContext> contextPool(ApplicationContext applicationContext) {
+        return new BeanPool<>(() -> applicationContext.createBean(ReactJSContext.class));
+    }
+
+    @Singleton
+    ApplicationEventListener<ReactJSSourcesChangedEvent> bookCleaner(BeanPool<ReactJSContext> contextPool) {
```

Comment:
`bookCleaner` -> `onSourcesChanged` ? Or `poolCleaner`

Reply:
`poolCleaner` not sure why I wrote that :D